### PR TITLE
[NEW] Subscription enabled marketplace

### PR DESCRIPTION
**Pull request by d-gubert**
This PR integrates Rocket.Chat with the new Marketplace-API to support apps subscriptions via marketplace.

It's dependant on https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/126

### CODE REVIEW

File: app/apps/server/bridges/users.js [hunk 1]
```diff
@@ -14,4 +17,14 @@ export class AppUserBridge {
...
 	}
+
+	async getActiveUserCount() {
+		return Users.find({
+			active: true,
+			$or: [
+				{ 'federation.peer': Federation.localIdentifier },
+				{ federation: { $exists: false } },
+			],
+		}).count();
```
Comment (geekgonecrazy) [inline app/apps/server/bridges/users.js]: @sampaiodiego is this going to be safe to call at any sort of interval?

Comment (sampaiodiego) [reply app/apps/server/bridges/users.js]: nope.. I was hoping to have a better alternative for this by tomorrow.. 

@alansikora when can we expect to have the new flag? :grimacing:

Comment (d-gubert) [reply app/apps/server/bridges/users.js]: If we can get a better way to do this before the release, would surely be best 😬

Comment (d-gubert) [reply app/apps/server/bridges/users.js]: @sampaiodiego @alansikora any news on this?

Comment (geekgonecrazy) [reply app/apps/server/bridges/users.js]: https://github.com/RocketChat/Rocket.Chat/pull/15004

Comment (sampaiodiego) [reply app/apps/server/bridges/users.js]: https://github.com/RocketChat/Rocket.Chat/pull/15004 has been merged! you can rely on `isRemote` field on `users` collection 😉

File: app/apps/server/orchestrator.js [hunk 2]
```diff
@@ -12,11 +12,10 @@ export let Apps;
 
 class AppServerOrchestrator {
 	constructor() {
-		if (Permissions) {
-			Permissions.createOrUpdate('manage-apps', ['admin']);
-		}
+		Permissions.createOrUpdate('manage-apps', ['admin']);
 
-		this._marketplaceUrl = 'https://marketplace.rocket.chat';
+		// TODO: Replace with production url
```
Comment (rodrigok) [inline app/apps/server/orchestrator.js]: Reminder: Missing this change

File: app/models/server/models/Users.js [hunk 3]
```diff
@@ -463,6 +463,10 @@ export class Users extends Base {
 		return this.find(query, options);
 	}
 
+	findActive() {
+		return this.find({ active: true });
```
Comment (rodrigok) [inline app/models/server/models/Users.js]: Missing the possibility to pass options

File: app/models/server/models/Users.js [hunk 4]
```diff
@@ -671,6 +675,10 @@ export class Users extends Base {
 		return this.find({ isRemote: true });
 	}
 
+	findActiveRemote() {
+		return this.find({ active: true, isRemote: true });
```
Comment (rodrigok) [inline app/models/server/models/Users.js]: Missing the possibility to pass options

File: app/apps/server/communication/rest.js [hunk 5]
```diff
@@ -7,6 +7,12 @@ import { getWorkspaceAccessToken, getUserCloudAccessToken } from '../../../cloud
 import { settings } from '../../../settings';
 import { Info } from '../../../utils';
 
+const getDefaultHeaders = () => ({
+	'X-Apps-Engine-Version': '1.4.2', // TODO: Info.marketplaceApiVersion,
```
Comment (rodrigok) [inline app/apps/server/communication/rest.js]: Reminder: Missing change

File: app/apps/server/communication/rest.js [hunk 6]
```diff
@@ -92,15 +98,23 @@ export class AppsRestApi {
...
+
 					const token = getUserCloudAccessToken(this.getLoggedInUser()._id, true, 'marketplace:purchase', false);
 					if (!token) {
 						return API.v1.failure({ error: 'Unauthorized' });
 					}
 
-					return API.v1.success({ url: `${ baseUrl }/apps/${ this.queryParams.appId }/buy?workspaceId=${ workspaceId }&token=${ token }` });
+					return API.v1.success({
+						url: `${ baseUrl }/apps/${ this.queryParams.appId }/${
+							this.queryParams.purchaseType === 'buy' ? this.queryParams.purchaseType : 'subscribe'
```
Comment (rodrigok) [inline app/apps/server/communication/rest.js]: Why need this check since it's already requiring the property to have only valid values on line 104?

Comment (d-gubert) [reply app/apps/server/communication/rest.js]: The `purchaseType` property from the app does _not_ map directly to the endpoint's path, so this is not exaclty a validation, more of a mapping

**Pull request merged**